ml-agents/mlagents/trainers/stats.py

```diff
@@ -35,6 +36,21 @@ def write_text(self, category: str, text: str, step: int) -> None:
         pass
 
 
+class GaugeWriter(StatsWriter):
+    """
+    Write all stats that we recieve to the timer gauges, so we can track them offline easily
+    """
+
+    def write_stats(
+        self, category: str, values: Dict[str, StatsSummary], step: int
+    ) -> None:
+        for val, stats_summary in values.items():
+            set_gauge(f"{category}.{val}.mean", float(stats_summary.mean))
```

Comment:
This currently produces gauges like
```
"ppo_3DBall.Policy/Entropy.mean": {
    "value": 1.4189382791519165,
    "min": 1.4189382791519165,
    "max": 1.4189382791519165,
    "count": 1
},
"ppo_3DBall.Policy/Learning Rate.mean": {
    "value": 0.00029642722802236676,
    "min": 0.00029642722802236676,
    "max": 0.00029642722802236676,
    "count": 1
},
"ppo_3DBall.Environment/Cumulative Reward.mean": {
    "value": 1.0899818162069077,
    "min": 1.0899818162069077,
    "max": 1.0899818162069077,
    "count": 1
},
"ppo_3DBall.Environment/Episode Length.mean": {
    "value": 20.899817850637522,
    "min": 20.899817850637522,
    "max": 20.899817850637522,
    "count": 1
},
"ppo_3DBall.Policy/Extrinsic Value Estimate.mean": {
    "value": -0.3887239396572113,
    "min": -0.3887239396572113,
    "max": -0.3887239396572113,
    "count": 1
},
"ppo_3DBall.Policy/Extrinsic Reward.mean": {
    "value": 1.088868589949434,
    "min": 1.088868589949434,
    "max": 1.088868589949434,
    "count": 1
}
```
which is good that they all show up, but a bit ugly in the naming. Any suggestions? I'm thinking about stripping spaces and replacing `/` with `.`

Reply:
@xiaomaogy I think this will record ELO like you asked about.

Reply:
On one hand I agree that these are ugly, but on the other this way keeps them identical to how they are written out by the trainer in case we needed to match them exactly in the future.  So, not sure what the best path would be here.

Reply:
This records the last min-max-mean right? So grabbing the "value" of the reward would be equivalent of grabbing the last reported reward. Might be useful @anupambhatnagar

Reply:
@xiaomaogy yep it will record ELO - actually it will record everything in Tensorboard. 

The category names (e.g. `Policy/Extrinsic Reward`) are currently structured like that so that they show up as proper categories in Tensorboard. I'm OK with stripping out / and space if needed. 

Would it be possible to make them children? e.g. rather than appending ppo_3DBall to every entry have a 
```
ppo_3DBall.stats: {
    Policy/Learning Rate.mean": {
    "value": 0.00029642722802236676,
    "min": 0.00029642722802236676,
    "max": 0.00029642722802236676,
    "count": 1
    },
etc.
}
```

Reply:
> This records the last min-max-mean right

The gauges record the min, max, and most recent value that has been passed to them. They're analogous to gauges in statsd and prometheus:
https://github.com/statsd/statsd/blob/master/docs/metric_types.md#gauges
https://prometheus.io/docs/concepts/metric_types/#gauge
So the value at the end of training is the average reward from the last summary interval. I only ran training for a few steps, so in this example, the min, max, and current value all look the same.

I didn't plan on stripping anything out from the tensorboard names, just the ones for the keys here.

I'd prefer to keep the gauges flat here, and go for a naming scheme like statsd (again), e.g. with `.` as the delimited. Some examples here https://matt.aimonetti.net/posts/2013-06-practical-guide-to-graphite-monitoring/#naming-metrics

Reply:
And just to be clear, the gauges aren't doing any averaging - that's just the mean from the StatsSummary.

Reply:
@chriselion This looks perfect! exactly what I need.